### PR TITLE
Create Echarts chart components

### PR DESCRIPTION
**Pull request by binzek**
Created the following Echarts chart components with the help of Evidence's implementation.
- AreaChart
- BarChart
- BubbleChart
- FunnelChart
- SankeyChart

### CODE REVIEW

File: src/ui/.env.local [hunk 1]
```diff
@@ -1 +0,0 @@
-NEXT_PUBLIC_JINJAT_URL=http://127.0.0.1:8581
```
Comment (buremba) [inline src/ui/.env.local:1]: let's not edit file

File: src/ui/src/components/chart/AreaChart.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import {
+  JinjatECharts,
+  JinjatEChartsProps,
+} from "@components/chart/JinjatECharts";
+
+interface JinjatAreaChartsProps extends JinjatEChartsProps {
+  data: undefined;
+  x: undefined;
```
Comment (buremba) [inline src/ui/src/components/chart/AreaChart.tsx:9]: let's add types for these arguments

File: src/ui/src/components/chart/BarChart.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import {
+  JinjatECharts,
+  JinjatEChartsProps,
+} from "@components/chart/JinjatECharts";
+
+interface JinjatBarChartsProps extends JinjatEChartsProps {
+  data: undefined;
+  x: undefined;
```
Comment (buremba) [inline src/ui/src/components/chart/BarChart.tsx:9]: same here, please add type hints

File: src/ui/src/components/chart/BubbleChart.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import {
+  JinjatECharts,
+  JinjatEChartsProps,
+} from "@components/chart/JinjatECharts";
+
+interface JinjatBubbleChartsProps extends JinjatEChartsProps {
+  data: undefined;
+  x: undefined;
+  y: undefined;
```
Comment (buremba) [inline src/ui/src/components/chart/BubbleChart.tsx:10]: same here, please add type hints

File: src/ui/src/components/chart/FunnelChart.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        `options.series` is not null, please use `ECharts` component instead.
+      </div>
+    );
+  }
+  options.series = [
+    {
+      type: "funnel",
+      encode: {
+        data: props.data,
+        nameCol: props.nameCol,
```
Comment (buremba) [inline src/ui/src/components/chart/FunnelChart.tsx:27]: are you sure that this is valid? I don't see the `nameCol` in Echarts config: https://echarts.apache.org/en/option.html#series-funnel.encode

File: src/ui/src/components/chart/JinjatECharts.tsx [hunk 6]
```diff
@@ -1,74 +1,79 @@
...
-            queryKey: ["_analysis", dataset],
-        }
-    });
+  dataset,
+  options = {},
+  theme,
+}) => {
+  // @ts-ignore
+  const isRenderable = memoize(
+    () =>
```
Comment (buremba) [inline src/ui/src/components/chart/JinjatECharts.tsx]: please revert the changes in this file as if you didn't change the code in this file.

File: src/ui/src/components/chart/SankeyChart.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        `options.series` is not null, please use `ECharts` component instead.
+      </div>
+    );
+  }
+  options.series = [
+    {
+      type: "sankey",
+      encode: {
+        data: props.data,
+        sourceCol: props.sourceCol,
```
Comment (buremba) [inline src/ui/src/components/chart/SankeyChart.tsx:28]: Same with funnel, does EChart support these configs?

**Review by buremba — CHANGES_REQUESTED** [review]

File: src/ui/.env.local (hunk 1 above)
Comment (binzek) [reply src/ui/.env.local:1]: So are there any other ways to connect the public URL of the server to the app instead of the self-hosted URL without editing this file?

File: src/ui/src/components/chart/AreaChart.tsx (hunk 2 above)
Comment (binzek) [reply src/ui/src/components/chart/AreaChart.tsx:9]: In Evidence's code, they all are given as undefined. I also had doubts about it. Is there any other source where I could find the actually expected types of these arguments?

File: src/ui/src/components/chart/SankeyChart.tsx (hunk 7 above)
Comment (wajid-nv) [reply src/ui/src/components/chart/SankeyChart.tsx:28]: I need clarification on this. I see these properties in the Evidence example, but as you said, not in the Echarts config. I am unsure about how can I find the actual property from Echarts API docs that match with the Evidence example.

File: src/ui/src/components/chart/FunnelChart.tsx (hunk 5 above)
Comment (binzek) [reply src/ui/src/components/chart/FunnelChart.tsx:27]: I need clarification on this. I see these properties in the Evidence example, but as you said, not in the Echarts config. I am unsure about how can I find the actual property from Echarts API docs that match with the Evidence example.

File: src/ui/src/components/chart/JinjatECharts.tsx (hunk 6 above)
Comment (binzek) [reply src/ui/src/components/chart/JinjatECharts.tsx]: Okay. That was just a formatting edit. Will revert it.
